packages/voice/src/util/Secretbox.ts

```diff
@@ -33,16 +63,57 @@ const libs = {
 			sodium.api.randombytes_buf(buffer);
 			return buffer;
 		},
+
+		crypto_aead_xchacha20poly1305_ietf_encrypt: (
+			message: Buffer,
+			additionalData: Buffer,
+			nonce: Buffer,
+			key: ArrayBufferLike,
+		) => {
+			return sodium.api.crypto_aead_xchacha20poly1305_ietf_encrypt(message, additionalData, null, nonce, key);
+		},
+		crypto_aead_xchacha20poly1305_ietf_decrypt: (
+			message: Buffer,
+			additionalData: Buffer,
+			nonce: Buffer,
+			key: ArrayBufferLike,
+		) => {
+			return sodium.api.crypto_aead_xchacha20poly1305_ietf_decrypt(message, additionalData, null, nonce, key);
+		},
 	}),
 	'libsodium-wrappers': (sodium: any): Methods => ({
 		open: sodium.crypto_secretbox_open_easy,
 		close: sodium.crypto_secretbox_easy,
 		random: sodium.randombytes_buf,
+
+		crypto_aead_xchacha20poly1305_ietf_encrypt: (
+			message: Buffer,
+			additionalData: Buffer,
+			nonce: Buffer,
+			key: ArrayBufferLike,
+		) => {
+			return sodium.crypto_aead_xchacha20poly1305_ietf_encrypt(message, additionalData, null, nonce, key);
+		},
+		crypto_aead_xchacha20poly1305_ietf_decrypt: (
+			message: Buffer,
+			additionalData: Buffer,
+			nonce: Buffer,
+			key: ArrayBufferLike,
+		) => {
+			return sodium.crypto_aead_xchacha20poly1305_ietf_decrypt(null, message, additionalData, nonce, key);
+		},
 	}),
 	tweetnacl: (tweetnacl: any): Methods => ({
 		open: tweetnacl.secretbox.open,
 		close: tweetnacl.secretbox,
 		random: tweetnacl.randomBytes,
+
+		crypto_aead_xchacha20poly1305_ietf_encrypt: () => {
```

Comment:
If tweetnacl doesn't support this, lets deprecate or remove it and release a major version for voice. or see if theres an alternative to tweetnacl